src/scrapers/isracard.ts

```diff
@@ -0,0 +1,234 @@
+
+      if (txnGroups) {
+        let txnIsrael: any[] = [];
+        let txnAbroad: any[] = [];
+        txnGroups.forEach((txnGroup: { txnIsrael: any; txnAbroad: any }) => {
+          if (txnGroup.txnIsrael) {
+            const txns = txnGroup.txnIsrael.map((txn: any) => {
+              return Object.assign(txn, {
+                processedDate: account.processedDate,
+              });
+            });
+            txnIsrael.push(...txns);
+          }
+
+          if (txnGroup.txnAbroad) {
+            const txns = txnGroup.txnAbroad.map((txn: any) => {
+              return Object.assign(txn, {
+                processedDate: account.processedDate,
+              });
+            });
+            txnAbroad.push(...txns);
+          }
+        });
+        const accountNum = lodash.get(account, 'accountNumber');
+        accountTxns[accountNum] = {
+          accountNumber: accountNum,
+          index: account.index,
+          txnIsrael: txnIsrael,
+          txnAbroad: txnAbroad,
+        };
+      }
+    });
+    return accountTxns;
+  }
+  return 0;
+}
+
+async function getMonthDashboard(
+  page: puppeteer.Page,
+  monthMoment: moment.Moment,
+  options?: isracardOptions
+) {
+  // get accounts data
+  const billingDate = monthMoment.format('YYYY-MM-DD');
+  const accountsUrl = `${SERVICE_URL}?reqName=DashboardMonth&actionCode=0&billingDate=${billingDate}&format=Json`;
+  const getDashboardFunction = fetchGetWithinPage<IsracardDashboardMonth>(
+    page,
+    accountsUrl
+  );
+
+  if (options && options.validateSchema) {
+    const data = await getDashboardFunction;
+    let validation = await validateSchema(isracardDashboardMonth, data);
+    return {
+      data,
+      ...validation,
+    };
+  } else {
+    return { data: await getDashboardFunction };
+  }
+}
+
+async function getMonthTransactions(
+  page: puppeteer.Page,
+  monthMoment: moment.Moment,
+  options?: isracardOptions
+) {
+  /* get transactions data */
+  const month = monthMoment.month() + 1;
+  const monthStr = month < 10 ? `0${month}` : month.toString();
+  const transUrl = `${SERVICE_URL}?reqName=CardsTransactionsList&month=${monthStr}&year=${monthMoment.year()}&requiredDate=N`;
+  const getTransactionsFunction = fetchGetWithinPage<
+    IsracardCardsTransactionsList
+  >(page, transUrl);
+
+  if (options && options.validateSchema) {
+    const data = await getTransactionsFunction;
+    let validation = await validateSchema(isracardCardsTransactionsList, data);
+    return {
+      data,
+      ...validation,
+    };
+  } else {
+    return { data: await getTransactionsFunction };
+  }
+}
+
+export async function isracard(
+  page: puppeteer.Page,
+  credentials: isracardCredentials,
+  options?: isracardOptions
+) {
+  const BASE_URL = 'https://digital.isracard.co.il';
+  await page.goto(`${BASE_URL}/personalarea/Login`);
+
+  await login(credentials, page);
+
+  /* dates logic  */
+  let startMoment = moment().subtract(1, 'months').startOf('month');
```

Comment:
for me, I just use `getMonthTransactions` so I don't need that logic to run before, so maybe you can move it to a function and call it only on the necessary functions?

Reply:
done